File: LodgeNET.SPA/src/app/_models/room.ts
import { Building } from "./building";

export interface Room {
    id: number;
    roomNumber: string;
    surgeCapacity: number;
    capacity: number;
    squareFootage: number;
    floor: number;
    buildingId: number;
    building?: Building;
    roomType?: string;
    currentGuestCount?: number;
}

File: LodgeNET.API/DAL/Models/Room.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace LodgeNET.API.DAL.Models
{
    public class Room
    {
        [Key]
        public int Id { get; set; }
        public string RoomNumber { get; set; }
        public int SurgeCapacity { get; set; }
        public int Capacity { get; set; }
        public int SquareFootage { get; set; }
        public int Floor { get; set; }
        public int BuildingId { get; set; }
        [ForeignKey("BuildingId")]
        public Building Building { get; set; }
        public string RoomType { get; set; }
    }
}